grammar.js

```diff
@@ -786,6 +787,7 @@ module.exports = grammar({
 
     type_binding: $ => seq(
       field('name', $._type_identifier),
+      field('type_arguments', optional($.type_arguments)),
```

Comment:
Should this by `type_parameters`, instead of `type_arguments`?

Reply:
I think `type_parameters` is not suitable here, because `type_parameters` is used at the position where new type parameters are defined (permits to write something like `T: Debug`), while `type_arguments` is used at the position where assigning type to them.

For example:
```text
fn foo<T: Debug, C, V>()
       ~~~~~~~~ type_parameter: defining new type parameter 
where
    V: Deref<Target = T> + Send,
    C: Bar<Baz<T> = V>,
               ~ type_argument: assigning type parameter     
{}
```

If `type_binding` accept `type_arguments`, this will be ok
```text
fn foo<T, C, V>()
where
    V: Deref<Target = T> + Send,
    C: Bar<Baz<T: Debug> = V>,
{}
```
which is not accepted by compiler.

Also, I just noticed that if `associated_type_bounds` feature is enabled, syntax like above is permitted. Do we want to include `associated_type_bounds` support to this PR?

Reply:
> type_parameters is used at the position where new type parameters are defined (permits to write something like `T: Debug`), while type_arguments is used at the position where assigning type to them.

Yes, I believe that on the *left* side of a `type_binding`, you are introducing new parameters. On the *right* side of the binding, you can refer to those type parameters. It's just like how in an associated type definition itself, there are type parameters on the left side of the `=` sign.

Reply:
> I believe that on the left side of a type_binding, you are introducing new parameters. 

We are referring the parameters on the left side too. We cannot introduce new parameters there. [playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=d04f6a3a7c7234489a17c49595f244cb)

```text
fn foo<T, C, V>()
where
    C: Bar<Baz<T> = V>,
               ~ This T is *referring* T which is defined on function signature.
{}
```

Reply:
The code used in test is not correct. I wanted to keep it minimal to check just syntax only, so there are missing type parameter definitions or import statement. I'm sorry if this made you get confused.

Reply:
Ah ok, thanks for the example.